link/uprobe.go

```diff
@@ -0,0 +1,275 @@
+package link
+
+import (
+	"debug/elf"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+func init() {
+	symbolsCache = internal.NewSymbolsCache()
+}
+
+var (
+	uprobeEventsPath = filepath.Join(tracefsPath, "uprobe_events")
+
+	// reUprobeSymbol is used to strip invalid characters from the uprobe symbol
+	// as they are not allowed to be used as the EVENT token in tracefs.
+	reUprobeSymbol = regexp.MustCompile("[^a-zA-Z0-9]+")
+
+	// TODO(matt): discuss if a cache is needed and if the current implementation
+	// is correct.
+	symbolsCache *internal.SymbolsCache
+)
+
+// Uprobe attaches the given eBPF program to a perf event that fires when the
+// given symbol starts executing in the given binary.
+// For example, /bin/bash::readline():
+//
+//	Uprobe("/bin/bash", "readline", prog)
+//
+// The resulting Link must be Closed during program shutdown to avoid leaking
+// system resources.
+//
+// Since the implementation of Uprobe is similar to the implementation of Kprobe,
```

Comment:
This is a comment on an exported function, and from a godoc perspective, `some functions in this file` is a bit meaningless. I would remove this paragraph.